Assertion failed on Python 3.14+

**Describe the bug**
CodeChecker fails due to a change in Python 3.14, discussed in:
https://github.com/python/cpython/issues/140922

When running an SCA build with CodeChecker on Zephyr, the following error occurs:

```
Traceback (most recent call last):
  File "/home/bas/Workspace/zephyrproject/.venv/lib64/python3.14/site-packages/codechecker_common/cli.py", line 259, in main
    sys.exit(args.func(args))
             ~~~~~~~~~^^^^^^
  File "/home/bas/Workspace/zephyrproject/.venv/lib64/python3.14/site-packages/codechecker_analyzer/cli/parse.py", line 658, in main
    report.trim_path_prefixes(trim_path_prefixes)
    ~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^
  File "/home/bas/Workspace/zephyrproject/.venv/lib64/python3.14/site-packages/codechecker_report_converter/report/__init__.py", line 388, in trim_path_prefixes
    self.file.trim(path_prefixes)
    ~~~~~~~~~~~~~~^^^^^^^^^^^^^^^
  File "/home/bas/Workspace/zephyrproject/.venv/lib64/python3.14/site-packages/codechecker_report_converter/report/__init__.py", line 105, in trim
    self.__path = util.trim_path_prefixes(
                  ~~~~~~~~~~~~~~~~~~~~~~~^
        self.__path, path_prefixes)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/bas/Workspace/zephyrproject/.venv/lib64/python3.14/site-packages/codechecker_report_converter/util.py", line 81, in trim_path_prefixes
    assert regex_str[-2:] == '\\Z', \
           ^^^^^^^^^^^^^^^^^^^^^^^
AssertionError: fnmatch.translate should leave \\Z at the end of the matcher!
```

**CodeChecker version**
6.27.1

**To Reproduce**
Run a Zephyr CodeChecker build as documented in: https://docs.zephyrproject.org/latest/develop/sca/codechecker.html#

**Expected behaviour**
The build should finish successfully.

**Desktop (please complete the following information)**
 - OS: Fedora 43 on x64 with Python 3.14.2
 - Browser: n/a
 - Version: n/a

**Additional context**
A workaround that resolved the issue locally was to replace `\\Z` with `\\z` in `util.py`.

--- a/tools/report-converter/codechecker_report_converter/util.py
+++ b/tools/report-converter/codechecker_report_converter/util.py
@@ -78,8 +78,9 @@ def trim_path_prefixes(path: str, prefixes: Optional[List[str]]) -> str:
             prefix += '/'
 
         regex_str = fnmatch.translate(prefix)
-        assert regex_str[-2:] == '\\Z', \
-               r'fnmatch.translate should leave \\Z at the end of the matcher!'
+        assert regex_str[-2:] in ('\\Z', '\\z'), \
+               r'fnmatch.translate should leave \\Z or \\z ' \
+               r'at the end of the matcher!'
 
         prefix_matcher = re.compile(regex_str[:-2])
 

